### ISSUE TITLE
Telegram2Logstash?

### ISSUE THREAD

**Issue by itismyusernameforgithub** [Programie/Telegram2Elastic#2]
and hello again, it's me :) I tried to use your development to send a message to elasticsearch via logstash

instead of the elasticsearch address, I put the logstash address (use http input), but I get an error:
`The client noticed that the server is not Elasticsearch and we do not support this unknown product`

if you use TCP (or HTTP) input, then the connection to the logstash is lost after 10 seconds and in the data received from telegram, the message field is null:

```
Accept-Encoding: identity
accept: application/vnd.elasticsearch+json; compatible-with=8
Host: 10.10.10.10:1234
connection: keep-alive
Content-Length: 137
content-type: application/vnd.elasticsearch+json; compatible-with=8
PUT /telegram/_doc/1 HTTP/1.1
user-agent: elasticsearch-py/8.3.3 (Python/3.9.7; elastic-transport/8.4.0)
x-elastic-client-meta: es=8.3.3,py=3.9.7,t=8.4.0,ur=1.26.12
{"timestamp":"2022-03-01T19:49:05+00:00","sender":{"username":null},"chat":"MyChat","message":null}
```

could you tell me how i can send messages from telegram to logstash with Telegram2Elastic? or you know any Telegram2Logstash development

**Comment by Programie** [comment]
As I'm simply using the official Elasticsearch Python module, which only supports talking to an Elasticsearch endpoint, there is no way in using something different between or instead of Elasticsearch like Logstash.

I guess the client looses the connection as it does not get the correct response from Logstash as it is not meant to be used as an Elasticsearch endpoint.

It shouldn't be too hard to also support other endpoints. I will have a look.

**Comment by Programie** [comment]
I've rewritten the whole code to allow adding different output types.

As of now, the following output types are implemented:

* Elasticsearch: Write to an Elasticsearch instance just like before
* File: Write per message JSON lines into any file (i.e. to be picked up by Filebeat or Logstash)
* TCP: Send messages as JSON string to any TCP socket (i.e. to be received using the "tcp" input in Logstash)

I just did some testing for the File and TCP outputs, but didn't test anything with Logstash or Elasticsearch yet. That's still something I have to do before releasing a new version.

**Comment by Programie** [comment]
Thanks for your answers.

1. Usually, it should work to specify "json" as codec in the tcp input:

```
input {
  tcp {
    port => 12345
    codec => json
  }
}
```

But I didn't test that yet.

2. Good point about the multiple hosts for the Elasticsearch connection. I will update the code to support multiple hosts.

3. I also already saw the issue about the AttributeError while reading messages from channels. I will fix it.

4. I already thought about renaming the project to something like "Telelog" or something like that as it will change into a way where it uses Telegram messages and send it to any output to log those messages.

**Comment by Programie** [comment]
Good news: I can reproduce the issue with the `listen` mode.

Bad news: I don't know what's wrong.

I've debugged the code down to the actual event handler which doesn't seem to be triggered at all. Adding a simple `print(event)` into the handler method doesn't print anything once a new message is available:

```python
async def handler(event):
    print(event)
```

Manually adding the event handler with `add_event_handler()` also has no effect. I guess I have to dig a bit deeper...

**Comment by Programie** [comment]
Found the issue quicker than I thought it would take.

Getting the loop from `asyncio.get_event_loop()` instead of `asyncio.new_event_loop()` fixes the issue.

I've changed that to `asyncio.new_event_loop()` as starting the event loop with a loop retrieved from `asyncio.get_event_loop()` results in showing deprecation warnings "There is no current event loop". But unfortunately, this resulted in not listening for events which is kind of weird.
